raiden/tests/integration/test_send_queued_messages.py

```diff
@@ -238,15 +239,16 @@ def test_payment_statuses_are_restored(  # pylint: disable=unused-argument
     with watch_for_unlock_failures(*raiden_network):
         waiting.wait_for_healthy(app0_restart, app1.address, network_wait)
 
-        waiting.wait_for_payment_balance(
-            raiden=app1,
-            token_network_registry_address=token_network_registry_address,
-            token_address=token_address,
-            partner_address=app0_restart.address,
-            target_address=Address(target_address),
-            target_balance=spent_amount,
-            retry_timeout=network_wait,
-        )
+        with gevent.Timeout(60):
```

Comment:
Just out of curiosity, why do we need the timeout here?

Reply:
Without this, a failure like the one fixed by this PR will keep the test stuck until the test runner is killed. This makes the test fail faster and will show the unlock failures that happen due to the context manager.